clients/ordering/src/components/cart.tsx

```diff
@@ -57,38 +53,70 @@ class Cart extends Component<State> {
                   <IconButton>
                     <DeleteIcon
                       onClick={() =>
-                        (this.props as any).dispatch({
-                          type: 'REMOVE_ITEM',
-                          id: item.id,
-                        })
+                        (this.props as any).dispatch(removeItem(item.id))
                       }
                     />
                   </IconButton>
                 </ListItemSecondaryAction>
               </ListItem>
             );
           })}
-        </List>
-        <List>
           <ListItem>
-            <ListItemText
-              primary={'Subtotal: $' + this.props.total.toFixed(2)}
-            />
+            <ListItemText primary={'Subtotal'} />
+            <Typography variant={'subtitle2'}>
+              {'$' + this.props.total.toFixed(2)}
+            </Typography>
           </ListItem>
           <ListItem>
-            <ListItemText
-              primary={'Tax: $' + (this.props.total * 0.09).toFixed(2)}
-              secondary={'~9%'}
-            />
+            <ListItemText primary={'Tax'} />
+            <Typography variant={'subtitle2'}>
+              {'$' + (this.props.total * 0.09).toFixed(2)}
+            </Typography>
           </ListItem>
           <ListItem>
-            <ListItemText
-              primary={
-                'Total: $' +
-                (this.props.total + this.props.total * 0.09).toFixed(2)
-              }
-            />
+            <ListItemText primary={'Total'} />
+            <Typography
+              variant={'subtitle2'}
+              style={{ fontWeight: 'bold', fontSize: 'large' }}
+            >
+              {'$' + (this.props.total + this.props.total * 0.09).toFixed(2)}
+            </Typography>
           </ListItem>
+          <Link to="/orders" style={{ textDecoration: 'none' }}>
+            <Button
+              style={{
+                whiteSpace: 'nowrap',
```

Comment:
Preferably use material-ui's style solution to style this instead of creating an inline style.

Reply:
![21](https://user-images.githubusercontent.com/43417133/56995362-2ea6d600-6b67-11e9-9398-1f08f35fc142.PNG)
![22](https://user-images.githubusercontent.com/43417133/56995363-2ea6d600-6b67-11e9-9e42-2a7a74fcbd4e.PNG)
Not sure how to implement styles on a class. Is this close?

Reply:
You have components that are using `createStyles` to define the styles and `withStyles` to bind the styles with the component. Essentially just copying those components.

Reply:
![23](https://user-images.githubusercontent.com/43417133/56995472-7b8aac80-6b67-11e9-83ce-7542ce628ad6.PNG)
Changed to this, but the line under render() is still not right.

Reply:
You need to wrap the component using the styles like `withStyles(styles)(MyComponent)`

Reply:
yeah, I have that

Reply:
I think this may have been a similar issue from last sprint, but I think you're gonna have to switch the order of `connect` and `withStyles` when wrapping the components.

Reply:
That part is working fine. This is where the error is.
![24](https://user-images.githubusercontent.com/43417133/56995875-99a4dc80-6b68-11e9-81d6-ffdfd2057f3e.PNG)

Reply:
Oh shoot. You're gonna have to define the types for the props in the component, so `Component<State>` should be like `Component<State, CheckoutProps>` and you'll need to define `CheckoutProps`.  It will be something like ```interface CheckoutProps extends WithStyles<typeof styles> { ... }```. You can look how I'm doing it in the seating client.

Reply:
yeah, I've tried that but it just created a bunch of other errors
![25](https://user-images.githubusercontent.com/43417133/56996263-8f371280-6b69-11e9-91b7-72b0d0e23266.PNG)

Reply:
The order of props and state might be flipped from what I said.

Reply:
tried that too, even more errors

Reply:
What are your errors?

Reply:
![26](https://user-images.githubusercontent.com/43417133/56996412-1b493a00-6b6a-11e9-89df-31ca826ec472.PNG)
![27](https://user-images.githubusercontent.com/43417133/56996417-1d12fd80-6b6a-11e9-9f57-a80e1cae4228.PNG)

Reply:
Looks like you need to define the types for `finalTotal` and `boughtItems` in your `CheckoutProps`. What are the error messages for `state` and `value`?

Reply:
finalTotal and boughtItems are from the state interface. do I need to define them in CheckoutProps again? All this just seemed much more complicated than just doing one or two inline styles.
![28](https://user-images.githubusercontent.com/43417133/56996578-9ad70900-6b6a-11e9-9e3a-002dd30c0fc8.PNG)

Reply:
Ok. Your `State` type was misleading. I was assuming you were describing the local state for that component and not the redux state you're taking in as props. The convention is usually `React.Component<ComponentProps, ComponentState>` To fix this, move all the things from `State` to `CheckoutProps` and rename  `State` in `React.Component<State>` to `React.Component<CheckoutProps>`.

Reply:
Ok, thanks. That seemed to work. I'll go through and change some of my classes to accommodate this and then I pushed back up.